### PR TITLE
add `region` field to balancer protocol

### LINKED ISSUE #1121: add region field to `M2BInit`
add a new field `region` to `M2BInit`, and add support in the Monolith

the monolith should derive the region from these places, in this order:
- `FLY_REGION` environment variable
- configuration, with default value "unknown"

### PR DESCRIPTION
**Pull request by dyc3**
closes #1121

### CODE REVIEW

File: crates/harness/src/monolith.rs [hunk 1]
```diff
@@ -85,7 +85,10 @@ impl Monolith {
                 loop {
                     let (stream, _) = _listener.accept().await.unwrap();
                     let mut ws = tokio_tungstenite::accept_async(stream).await.unwrap();
-                    let init = M2BInit { port: http_port };
+                    let init = M2BInit {
+                        port: http_port,
+                        region: "unknown".into(),
```
Comment (cjrkoa) [inline crates/harness/src/monolith.rs:90]: what does .into() do?

File: crates/ott-balancer-protocol/src/monolith.rs [hunk 2]
```diff
@@ -93,6 +93,7 @@ pub enum MsgM2B {
 pub struct M2BInit {
     /// The port that the monolith is listening for HTTP requests on.
     pub port: u16,
+    pub region: String,
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
```
Comment (cjrkoa) [inline crates/ott-balancer-protocol/src/monolith.rs:99]: line 99?

File: server/balancer.ts
Comment (cjrkoa) [inline server/balancer.ts:1]: This file won't let me comment on individual lines so:
wss.on()? 
this.emit() - emits to what?

File: crates/harness/src/monolith.rs (hunk 1 above)
Comment (dyc3) [reply crates/harness/src/monolith.rs:90]: It comes from the `Into` trait. https://doc.rust-lang.org/std/convert/trait.Into.html

You can read more in depth about `From`/`Into` here: https://practice.rs/type-conversions/from-into.html Specifically, this `into()` converts a `&'static str` into a `String`.

File: server/ott-config.ts [hunk 3]
```diff
@@ -513,6 +519,11 @@ function postProcessConfig(): void {
 	if (conf.get("mail.enabled")) {
 		validateMail();
```
Comment (cjrkoa) [inline server/ott-config.ts:520]: what does mail mean in this context?

**Review by cjrkoa — APPROVED** [review]

File: server/balancer.ts
Comment (dyc3) [reply server/balancer.ts:1]: wss is a `WebSocket.Server` from the `ws` node js package.

`on(event, handler)` runs `handler` whenever `event` is emitted by the object.

`this` in that context is `BalancerManager`, which has an event bus that emits events from `BalancerManagerEvemts` (which ive just realized I've typod)

File: server/ott-config.ts (hunk 3 above)
Comment (dyc3) [reply server/ott-config.ts:520]: email. used for password recovery emails.

File: crates/ott-balancer-protocol/src/monolith.rs (hunk 2 above)
Comment (dyc3) [reply crates/ott-balancer-protocol/src/monolith.rs:99]: `derive` invokes derive macros. They are macros that automatically implement traits for you. For example, `#[derive(Clone)]` will automatically implement `Clone`, and allow you to call `clone()` on the type that it's implemented for.

**Pull request merged**